### PR TITLE
Rt1060 am457 bt support

### PR DESCRIPTION
**Pull request by MarkWangChinese**
with the three PRs: https://github.com/zephyrproject-rtos/zephyr/pull/31583, https://github.com/zephyrproject-rtos/zephyr/pull/35536 and this PR, the a2dp samples can run in the RT1060+ AW-AM457-D-uSD

### CODE REVIEW

File: drivers/serial/uart_mcux_lpuart.c [hunk 1]
```diff
@@ -356,7 +356,11 @@ static int mcux_lpuart_init(const struct device *dev)
 	uart_api_config->stop_bits = UART_CFG_STOP_BITS_1;
 	uart_api_config->data_bits = UART_CFG_DATA_BITS_8;
 	uart_api_config->flow_ctrl = UART_CFG_FLOW_CTRL_NONE;
-
+#if defined (CONFIG_BT_UART_ON_DEV_NAME)
+	if (strcmp(CONFIG_BT_UART_ON_DEV_NAME, dev->name) == 0) {
+		uart_api_config->flow_ctrl = UART_CFG_FLOW_CTRL_RTS_CTS;
```
Comment (MaureenHelm) [inline drivers/serial/uart_mcux_lpuart.c:361]: Flow control should be implemented using the dts property, `hw-flow-control`, similar to `drivers/serial/uart_mcux.c`. We should not use `CONFIG_BT_UART_ON_DEV_NAME` here.

File: boards/arm/mimxrt1060_evk/pinmux.c [hunk 2]
```diff
@@ -144,6 +144,25 @@ static int mimxrt1060_evk_init(const struct device *dev)
 			    IOMUXC_SW_PAD_CTL_PAD_PKE_MASK |
 			    IOMUXC_SW_PAD_CTL_PAD_SPEED(2) |
 			    IOMUXC_SW_PAD_CTL_PAD_DSE(6));
+
+#if defined (CONFIG_BT_UART_ON_DEV_NAME)
+	if (strcmp(CONFIG_BT_UART_ON_DEV_NAME, "UART_3") == 0) {
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1060_evk/pinmux.c:149]: This should use the dts property `hw-flow-control` instead of `CONFIG_BT_UART_ON_DEV_NAME`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks @MarkWangChinese 

Please review and address the compliance check failure.

File: drivers/serial/uart_mcux_lpuart.c (hunk 1 above)
Comment (MarkWangChinese) [reply drivers/serial/uart_mcux_lpuart.c]: Thanks @MaureenHelm , I have upated it, please help check it again.

File: boards/arm/mimxrt1060_evk/pinmux.c (hunk 2 above)
Comment (MarkWangChinese) [reply boards/arm/mimxrt1060_evk/pinmux.c]: Thanks @MaureenHelm , I have upated it, please help check it again.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Code changes look good now, but you still need to address the compliance check failure:
```
Error: See docs.zephyrproject.org/latest/contribute/#commit-guidelines for more details

Commit 04723e69b2:
1: UC6 Body has no content, should at least have 1 line.

Commit 1a3d406117:
1: UC6 Body has no content, should at least have 1 line.
```

**Comment by MarkWangChinese** [comment]
Hi @MaureenHelm, I have updated it, there are still some fails in buildkite/zephyr checking. It seems not related to my changes, please help confirm it. Thanks.

File: drivers/serial/uart_mcux_lpuart.c [hunk 3]
```diff
@@ -426,7 +427,9 @@ static const struct mcux_lpuart_config mcux_lpuart_##n##_config = {	\
 	.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(n)),		\
 	.clock_subsys = (clock_control_subsys_t)DT_INST_CLOCKS_CELL(n, name),\
 	.baud_rate = DT_INST_PROP(n, current_speed),			\
-	IRQ_FUNC_INIT							\
+	IRQ_FUNC_INIT,							\
```
Comment (dleach02) [inline drivers/serial/uart_mcux_lpuart.c:430]: The error is related to the comma after IRQ_FUNC_INIT. I suspect the expansion on the particular build to get to this probably already includes a comma.

From log file:

 /workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:430:15: error: expected expression before ',' token

  430 |  IRQ_FUNC_INIT,       \

      |               ^

../../../../../../../include/sys/util_internal.h:67:26: note: in definition of macro '__DEBRACKET'

   67 | #define __DEBRACKET(...) __VA_ARGS__

      |                          ^~~~~~~~~~~

../../../../../../../include/sys/util_internal.h:59:2: note: in expansion of macro '__GET_ARG2_DEBRACKET'

   59 |  __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)

      |  ^~~~~~~~~~~~~~~~~~~~

../../../../../../../include/sys/util_internal.h:54:2: note: in expansion of macro '__COND_CODE'

   54 |  __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)

      |  ^~~~~~~~~~~

../../../../../../../include/sys/util_macro.h:117:2: note: in expansion of macro 'Z_COND_CODE_1'

  117 |  Z_COND_CODE_1(_flag, _if_1_code, _else_code)

      |  ^~~~~~~~~~~~~

../../../../../../../include/devicetree.h:2211:2: note: in expansion of macro 'COND_CODE_1'

 2211 |  COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \

      |  ^~~~~~~~~~~

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:421:2: note: in expansion of macro 'LPUART_MCUX_DECLARE_CFG'

  421 |  LPUART_MCUX_DECLARE_CFG(n, LPUART_MCUX_IRQ_CFG_FUNC_INIT)

      |  ^~~~~~~~~~~~~~~~~~~~~~~

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:452:2: note: in expansion of macro 'LPUART_MCUX_INIT_CFG'

  452 |  LPUART_MCUX_INIT_CFG(n);

      |  ^~~~~~~~~~~~~~~~~~~~

zephyr/include/generated/devicetree_unfixed.h:9163:53: note: in expansion of macro 'LPUART_MCUX_INIT'

 9163 | #define DT_FOREACH_OKAY_INST_nxp_kinetis_lpuart(fn) fn(0)

      |                                                     ^~

../../../../../../../include/sys/util_internal.h:85:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_kinetis_lpuart'

   85 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__

      |                                    ^

/workdir/zephyr/drivers/serial/uart_mcux_lpuart.c:454:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'

  454 | DT_INST_FOREACH_STATUS_OKAY(LPUART_MCUX_INIT)

      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~

**Review by dleach02 — CHANGES_REQUESTED** [review]

**Comment by dleach02** [comment]
@MarkWangChinese do you still want to pursue this PR? There are build errors that need to be addressed.

File: drivers/serial/uart_mcux_lpuart.c (hunk 3 above)
Comment (MarkWangChinese) [reply drivers/serial/uart_mcux_lpuart.c]: Thanks, I switch the order of flow_ctrl and irq_config_func to put IRQ_FUNC_INIT in the last.

**Review by dcpleung — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**